Cannot build developer admin UI with gradle in v0.27.2-server

## Situation
Do following:
```
git checkout v0.27.2-server

./gradlew buildDeveloperAdminUI
```
This will fail with:

```
> Task :sechub-developertools:buildDeveloperAdminUI FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':sechub-developertools:buildDeveloperAdminUI'.
> Cannot expand ZIP '/home/albert/develop/projects/sechub/sechub-open-source/sechub-commons-model/build/libs/sechub-commons-model-0.27.2.jar' as it does not exist.

```
### First analyse
The problem is that we provide own version numbers for commons-model and commons-commons-core (so we can update and fix them independently from server releases)


## Wanted 
Build without problems...

@sven-dmlr :  A workaround is to use the current development branch (results in 0.0.0 which works always...) We got the problems only when we got the server version tag.

--- a/build-maven.gradle
+++ b/build-maven.gradle
@@ -19,13 +19,20 @@ subprojects {
         return;
     }
 
+    // we always set the libraries version in this case
+    // If we publish and the version is set, it means the dedicated, published lib version
+    // If not defined, the library version is 0.0.0 which represents a snapshot -
+    // the snapshot itself is related to the spring boot application (server or PDS) which
+    // contains the lib (e.g. commons-core, commons-model). So it is clear there
+    // could be changes to last published one.
+    version = VersionData.getLibrariesVersion()
+    
     /* ensure we are at GITHUB actions when we want to publish the libraries */
     if (! buildDoneByGitHubActions ){
         println("library $project.name publish will be ignored - only provided by github actions")
         return;
     }
     
-    version = VersionData.getLibrariesVersion()
 
 	apply plugin: 'maven-publish'
 	

--- a/sechub-developertools/build.gradle
+++ b/sechub-developertools/build.gradle
@@ -5,11 +5,14 @@
  * Root build file: "${rootProject.projectDir}/build.gradle"
  * ============================================================================
  */
+plugins {
+    id 'java-library'
+}
 dependencies {
 
-    compile project(':sechub-testframework')
-    compile project(':sechub-integrationtest')
-    compile project(':sechub-scan')
+    implementation project(':sechub-testframework')
+    implementation project(':sechub-integrationtest')
+    implementation project(':sechub-scan')
     
     implementation project(':sechub-commons-pds')
     
@@ -32,7 +35,7 @@ task startIntegrationTestAdminUI(type: JavaExec){
 
 }
 
-task buildDeveloperAdminUI(type: Jar) {
+task buildDeveloperAdminUI(type: Jar, dependsOn: build) {
     group 'sechub'
     description 'Builds the SecHub Developer Admin tool as standalone executable jar. Use launch-developer-admin-ui script to execute'
     baseName = 'sechub-developer-admin-ui'
@@ -43,7 +46,7 @@ task buildDeveloperAdminUI(type: Jar) {
     
     
     from { 
-        configurations.compile.collect { it.isDirectory() ? it : zipTree(it) } 
+        configurations.runtimeClasspath.collect { it.isDirectory() ? it : zipTree(it) } 
     }
 
     with jar

--- a/sechub-integrationtest/build.gradle
+++ b/sechub-integrationtest/build.gradle
@@ -5,15 +5,19 @@
  * Root build file: "${rootProject.projectDir}/build.gradle"
  * ============================================================================
  */
+plugins {
+    id 'java'
+    id 'java-library'
+}
 dependencies {
 
-    compile project(':sechub-testframework')
-    compile project(':sechub-adapter') // we reuse trustall approach from adapters
-    compile project(':sechub-shared-kernel') 
-    compile project(':sechub-scan-product-pds') // we want configuration entries available
-    compile library.springboot_starter_mail // to have easier access to email parts
+    api project(':sechub-testframework')
+    api project(':sechub-adapter') // we reuse trustall approach from adapters
+    api project(':sechub-shared-kernel') 
+    api project(':sechub-scan-product-pds') // we want configuration entries available
+    api library.springboot_starter_mail // to have easier access to email parts
     
-    implementation project(':sechub-commons-pds')
+    api project(':sechub-commons-pds')
 }
 import java.nio.file.Files;
 import java.nio.file.Paths;

--- a/sechub-shared-kernel/build.gradle
+++ b/sechub-shared-kernel/build.gradle
@@ -5,9 +5,13 @@
  * Root build file: "${rootProject.projectDir}/build.gradle"
  * ============================================================================
  */
+plugins {
+    id 'java-library'
+}
+ 
 dependencies {
 
-    compile project(':sechub-commons-model')
+    api project(':sechub-commons-model')
     compile project(':sechub-storage-sharedvolume-spring')
     compile project(':sechub-storage-s3-aws')
     compile project(':sechub-adapter')

